tools/azsdk-cli/Azure.Sdk.Tools.Cli/Services/APIView/APIViewAuthenticationService.cs

```diff
@@ -0,0 +1,260 @@
+        string environment = "production")
+    {
+        if (!string.IsNullOrEmpty(providedToken))
+        {
+            return providedToken;
+        }
+
+        string? githubToken = GetGitHubToken();
+        if (!string.IsNullOrEmpty(githubToken))
+        {
+            _logger.LogDebug("Using GitHub token for authentication");
+            return githubToken;
+        }
+
+        try
+        {
+            DefaultAzureCredential credential = new();
+            string scope = APIViewConfiguration.ApiViewScopes[environment];
+            
+            if (string.IsNullOrEmpty(scope))
+            {
+                _logger.LogWarning("Environment not valid.");
+                return null;
+            }
+
+            try
+            {
+                TokenRequestContext tokenRequest = new([scope]);
+                AccessToken tokenResponse = await credential.GetTokenAsync(tokenRequest);
+
+                _logger.LogInformation("Successfully obtained Azure token with scope {Scope}", scope);
+                return tokenResponse.Token;
+            }
+            catch (Exception ex)
+            {
+                _logger.LogWarning(ex, "Failed to get Azure token with scope {Scope}: {Message}", scope, ex.Message);
+            }
+        }
+        catch (Exception ex)
+        {
+            _logger.LogDebug(ex, "Failed to get any Azure token, no authentication available");
+        }
+
+        _logger.LogWarning(
+            "No authentication token available. APIView requests may fail if authentication is required.");
+        return null;
+    }
+
+    public async Task ConfigureAuthenticationAsync(HttpClient httpClient, string environment = "production",
+        string? providedToken = null)
+    {
+        string? token = await GetAuthenticationTokenAsync(providedToken, environment);
+        if (!string.IsNullOrEmpty(token))
+        {
+            httpClient.DefaultRequestHeaders.Authorization =
+                new AuthenticationHeaderValue("Bearer", token);
+
+            httpClient.DefaultRequestHeaders.Add("X-API-Key", token);
+        }
+
+        httpClient.DefaultRequestHeaders.Add("User-Agent", APIViewConfiguration.UserAgent);
+        httpClient.DefaultRequestHeaders.Add("Accept", "application/json");
+
+        _logger.LogDebug("Request headers configured - Authorization: {HasAuth}, User-Agent: {UserAgent}",
+            !string.IsNullOrEmpty(token) ? "Bearer [TOKEN]" : "None", APIViewConfiguration.UserAgent);
+    }
+
+    public string GetAuthenticationGuidance()
+    {
+        return @"Authentication is required to access APIView data. Please choose one of these options:
+
+1. **GitHub Token**:
+   - Set environment variable: GITHUB_TOKEN=your_token_here
+   - Or run: gh auth login
+
+2. **Azure CLI**:
+   - Run: az login
+   - Ensure you have access to APIView resources
+
+3. **Provide token directly**:
+   - Pass authToken parameter to the MCP tool
```

Comment:
FYI @praveenkuttappan something we'll need to sanitize for telemetry that logs parameters. Or possibly we should have a way to mark parameters as secret so they don't get logged.

Reply:
Just curious, why cant we use chainedTokenCredential instead of the PAT here, doesn't seem like a secure way to authenticate?

Reply:
Here is more information on token acquisition for building secure MCP servers https://eng.ms/docs/initiatives/ai-guidance-for-microsoft-developers/mcp/building-secure-mcp